### ISSUE TITLE
Can I enable streaming? Please help

### ISSUE THREAD

**Issue by JaDenis** [emirsahin1/llm-axe#18]
Hi, first of all I'd like to say this is a great library!
Exactly what I was thinking of doing myself.

Is there anyway I can enable streaming, so that responses are streamed? I'm particularly interested in function calling with streaming

**Comment by emirsahin1** [comment]
Hey! Thank you for the kind words.
Others have asked for this as well but I never had the time to implement it. I will implement it this week, I promise!
I'll update you here when a new update is released. Thanks again for your interest.

**Comment by JaDenis** [comment]
Thank you, dude!
This might seem like a zero value feature, but it is very important if you test on real people.
for users streaming is an UX thing and for us developers streaming can be utilized to check chunks and cancel if it clearly failed to follow instructions. I had countless times where first llm call went a little sideways, so that following calls produced irrelevant stuff.

**Comment by emirsahin1** [comment]
Of course, I totally get it and its been something I've wanted to add in for some time.
I just finished implementing and it should be in the latest release: 1.1.9. Yay!

Here is an example on how to use streaming:

```python
llm = OllamaChat(model="llama3.1")
ag = Agent(llm, custom_system_prompt="", stream=True)
res = ag.ask("Explain who you are in long detail")

for chunk in res:
    print(chunk, end="")
```

Let me know if you have any issues!
Thanks!

**Comment by JaDenis** [comment]
You the man!
After pip uninstall llm_axe, pip install llm_axe it works!

It seems to print line by line, not token by token. It absolutely covers my usecase for checking chunks before the final answer, but from UX standpoint feels like something is not right. It would be magical if you could look into this.

If it helps: on Christmas night I tried 'lolcat' library to make my llm responses pretty and similarly it printed response line by line with lolcat enabled, instead of token by token.

Thank you, again! You are doing a great service to the community!

**Comment by emirsahin1** [comment]
If you enable flushing when printing, that should fix it.
 like this:
```python
for chunk in res:
    print(chunk, end="", flush=True)
```

**Comment by JaDenis** [comment]
Thank you, so much! Should have tried it, before posting!
Ave Deus Mechanicus, dude!
